### ISSUE TITLE
no-unnecessary-whitespace and enforce-consistent-line-wrapping contradict each other

### ISSUE THREAD

**Issue by esafak** [sergioazoc/oxlint-tailwindcss#14]
## Bug

`no-unnecessary-whitespace` flags the whitespace that `enforce-consistent-line-wrapping` with `classesPerLine` produces, creating an unfixable cycle. Migrated from `eslint-plugin-better-tailwindcss`, which supported `preferSingleLine` and `group` options for handling multiline template literals.

## Reproduction

Minimal repro: https://github.com/esafak/oxlint-tailwindcss-repro

`.oxlintrc.json`:
```json
{
  "jsPlugins": ["oxlint-tailwindcss"],
  "settings": { "tailwindcss": { "entryPoint": "src/app.css" } },
  "rules": {
    "tailwindcss/no-unnecessary-whitespace": "warn",
    "tailwindcss/enforce-consistent-line-wrapping": ["error", { "printWidth": 100, "classesPerLine": 3 }]
  }
}
```

`src/example.svelte`:
```svelte
<script lang="ts">
    const classes = `bg-primary text-primary-foreground hover:bg-primary-hover focus:ring-2 focus:ring-ring-focus focus-visible:ring-2 focus-visible:ring-ring-focus disabled:bg-primary-disabled`;
</script>
<div class={classes}></div>
```

## Steps — the unfixable cycle

**Step 1:** `pnpm exec oxlint src` — 0 warnings, 1 error (`classesPerLine` violation)

**Step 2:** `pnpm exec oxlint src --fix` — autofix splits into 3 classes per line:
```
`bg-primary text-primary-foreground hover:bg-primary-hover
                    focus:ring-2 focus:ring-ring-focus focus-visible:ring-2
                    focus-visible:ring-ring-focus disabled:bg-primary-disabled`
```

**Step 3:** `pnpm exec oxlint src` — 1 warning (`no-unnecessary-whitespace`) + 2 errors. `no-unnecessary-whitespace` flags the indentation/newlines that `classesPerLine` just produced.

**Step 4:** `pnpm exec oxlint src --fix` — `no-unnecessary-whitespace` autofix collapses back to single line. Returns to step 1.

## Expected

`no-unnecessary-whitespace` should not flag whitespace that `enforce-consistent-line-wrapping` with `classesPerLine` produces. The rules are complementary in intent but contradict in practice because `no-unnecessary-whitespace` does not distinguish between accidental whitespace and intentional formatting from the companion rule.

**Comment by sergioazoc** [comment]
Confirmed and fixed in [3752776](https://github.com/sergioazoc/oxlint-tailwindcss/commit/3752776) — will ship in **v0.7.0**. Thanks for the precise repro and the four-step trace, that pinned the cycle exactly.

**Root cause:** `no-unnecessary-whitespace` was running `value.replace(/\s+/g, ' ')`, which doesn't distinguish between accidental whitespace (`flex   items-center`) and intentional formatting (`\n` + indent introduced by `enforce-consistent-line-wrapping`'s `classesPerLine` autofix).

**Fix:** the rule is now multiline-aware. It collapses runs of `[ \t]+` only within each line, leaving `\n` and the indentation that immediately follows them untouched. Two integration tests pin the cycle break: step 1 (line-wrapping autofix produces the expected multiline output) and step 2 (`no-unnecessary-whitespace` reports nothing on that output).

**Bonus:** while auditing for similar conflicts, I found ~12 other autofixing rules that had the same problem (sort, canonical, variant-order, important-position, logical/physical, var-syntax, negative-arb, no-unnec-arb, no-deprecated, no-duplicate, no-unknown suggestions, shorthand). They all did `splitClasses(value).join(' ')` to rebuild the class string, which collapsed multiline indentation just like the original bug. A new pair of helpers (`splitClassesWithSeparators` + `rebuildClassString`) preserves separators verbatim for 1-to-1 transforms and degrades gracefully for length-changing ones. So `classesPerLine` is now compatible with autofix across the entire plugin, not just this rule.
